pkg/commands/exec_live_default.go

```diff
@@ -0,0 +1,122 @@
+// +build !windows
+
+package commands
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"unicode/utf8"
+
+	"github.com/kr/pty"
+	"github.com/mgutz/str"
+)
+
+// RunCommandWithOutputLiveWrapper runs a command and return every word that gets written in stdout
+// Output is a function that executes by every word that gets read by bufio
+// As return of output you need to give a string that will be written to stdin
+// NOTE: If the return data is empty it won't written anything to stdin
+// NOTE: You don't have to include a enter in the return data this function will do that for you
+func RunCommandWithOutputLiveWrapper(c *OSCommand, command string, output func(string) string) (errorMessage string, codeError error) {
+	cmdOutput := []string{}
+
+	splitCmd := str.ToArgv(command)
+	cmd := exec.Command(splitCmd[0], splitCmd[1:]...)
+
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "LANG=en_US.utf8", "LC_ALL=en_US.UTF-8")
+
+	tty, err := pty.Start(cmd)
+
+	if err != nil {
+		return errorMessage, err
+	}
+
+	stopAsking := make(chan struct{})
+
+	var waitForBufio sync.WaitGroup
+	waitForBufio.Add(1)
+
+	defer func() {
+		_ = tty.Close()
+	}()
+
+	go func() {
+		scanner := bufio.NewScanner(tty)
+		scanner.Split(scanWordsWithNewLines)
+		for scanner.Scan() {
+			select {
+			case <-stopAsking:
+				// just do nothing
+			default:
+				toOutput := strings.Trim(scanner.Text(), " ")
+				cmdOutput = append(cmdOutput, toOutput)
+				toWrite := output(toOutput)
+				if len(toWrite) > 0 {
+					_, _ = tty.Write([]byte(toWrite + "\n"))
+				}
+			}
+		}
+		waitForBufio.Done()
+	}()
+
+	err = cmd.Wait()
+	go func() {
+		stopAsking <- struct{}{}
+	}()
+	<-stopAsking
```

Comment:
I'm not sure what the expected flow here is. We create a channel, then after calling cmd.Wait we immediately write to the channel something to be read straight afterwards, and in the scanner. Scan loop nothing happens if we read the empty struct from the channel. I'm not sure that this is the right approach, but I'm also not sure what the initial problem was. Could you walk me through this?

Reply:
Ow thx for reminding me this is indeed stupid  
It's now removed :)  
In my head i though i still need to change a variable that might cause data races but that variable is now this channel